core/impl/src/main/java/com/blazebit/persistence/impl/function/every/FallbackEveryFunction.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.every;
+
+import com.blazebit.persistence.spi.FunctionRenderContext;
+
+/**
+ *
+ * @author Jan-Willem Gmelig Meyling
+ * @author Sayra Ranjha
+ * @since 1.4.0
+ */
+public class FallbackEveryFunction extends EveryFunction {
+
+    @Override
+    public void render(FunctionRenderContext context) {
+        renderLhs(context);
+        renderRhs(context);
+    }
+
+    public void renderLhs(FunctionRenderContext context) {
+        context.addChunk("SUM(CASE WHEN ");
```

Comment:
Since `false` is typically smaller than `true` this function can be implemented as `MIN(argument)` instead which will also preserve the underlying type which might be `BIT`, `NUMBER(1)`, `CHAR(1)` etc. This should fix the Oracle and MSSQL issue: https://travis-ci.org/Blazebit/blaze-persistence/jobs/576190641 and https://travis-ci.org/Blazebit/blaze-persistence/jobs/576190639